atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -244,6 +256,12 @@ long getId() {
         private final Optional<Long> id;
         private final long timestamp;
 
+        /*
+         * Very much of an edge case, but there is theoretically a risk of a wrong interpretation, if the byte
```

Comment:
That's a neat idea! Also put a test case to show that Jeremy's example now works correctly. (obviously the assumption is that we will not have negative bounds)